src/Akeneo/Connectivity/Connection/back/Infrastructure/Apps/Security/ScopeMapperRegistry.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Infrastructure\Apps\Security;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class ScopeMapperRegistry
+{
+    /** @var array<string, ScopeMapperInterface> */
+    private iterable $scopeMappers = [];
+
+    public function __construct(iterable $scopeMappers)
+    {
+        foreach ($scopeMappers as $scopeMapper) {
+            if (!$scopeMapper instanceof ScopeMapperInterface) {
+                throw new \InvalidArgumentException(
+                    \sprintf(
+                        '%s needs only %s',
+                        self::class,
+                        ScopeMapperInterface::class
+                    )
+                );
+            }
+            foreach ($scopeMapper->getScopes() as $scope) {
+                if (\array_key_exists($scope, $this->scopeMappers)) {
+                    throw new \InvalidArgumentException(
+                        sprintf(
+                            'The scope "%s" is already supported by the scope mapper "%s".',
+                            $scope,
+                            get_class($this->scopeMappers[$scope])
+                        )
+                    );
+                }
+                $this->scopeMappers[$scope] = $scopeMapper;
+            }
+        }
+    }
+
+    public function getAllScopes(): array
+    {
+        return \array_keys($this->scopeMappers);
+    }
+
+    /**
+     * Provides all information needed to display scopes.
+     * Filters the scopes by keeping the higher hierarchy scope.
+     *
+     * @param string[] $scopeList
+     *
+     * @throw \LogicArgumentException if a scope you provide does not exist
```

Comment:
```suggestion
     * @throw \LogicArgumentException if the given scope does not exist
```
so we won't confuse provide with providers